ui/src/components/admin/leadershipTeam/AdminLeadershipTeamContainer.js

```diff
@@ -1,125 +1,368 @@
 
-  useEffect(() => {
-    if (api) {
-      if (teams && teams.length) {
-        api.hideOverlay();
-      } else {
-        api.showLoadingOverlay();
+  const duplicateHandler = async (e) => {
+    setIsLoading(true);
+    try {
+      const { data, status } = await axios.get('/api/leadership-team/get', {
+        params: { id: e.target.value },
+      });
+
+      if (status !== 200) {
+        toast({
+          description:
+            'There was an issue with the request, please talk to a t3ch support',
+          status: 'warning',
+          duration: 8000,
+          isClosable: true,
+        });
       }
-    }
-  }, [teams, api]);
-
-  // Ag-Grid Helpers
-  // Formatters
-  const dateFormatter = (dateStr) => {
-    if (dateStr) {
-      const dateObj = DateTime.fromFormat(dateStr, dateFromFormat);
-      return dateObj.toFormat(dateToFormat);
-    }
 
-    return '';
-  };
+      setId('');
+      setSeasonFrom(data[0].seasonFrom);
+      setSeasonTo(data[0].seasonTo);
+      setCampus(data[0].campus);
+      setLifestage(data[0].lifestage);
+      setLifeGroup(data[0].lifeGroup);
+      setLeaders(data[0].leaders);
+      setDeleted(data[0].isDeleted);
+
+      setIsLoading(false);
+    } catch (err) {
+      console.log(err);
 
-  // Undo and Redo Functions
-  const undo = () => {
-    if (api) api.undoCellEditing();
+      toast({
+        description:
+          'There was an issue with the request, please talk to a t3ch support',
+        status: 'warning',
+        duration: 8000,
+        isClosable: true,
+      });
+      setIsLoading(false);
+    }
   };
 
-  const redo = () => {
-    if (api) api.redoCellEditing();
+  const resetHandler = () => {
+    setId('');
+    setSeasonFrom('');
+    setSeasonTo('');
+    setCampus('');
+    setLifestage('');
+    setLifeGroup('');
+    setLeaders('');
+    setDeleted(false);
+    setSelected();
+    setIsLoading(false);
   };
 
-  // Ag-Grid Column Definitions
-  const columnDefs = [];
+  const onSubmit = async (event) => {
+    event.preventDefault();
+
+    if (!lifeGroup || lifeGroup.length === 0) return;
+
+    setIsLoading(true);
+    let success = false;
+
+    if (id && id.length > 0) {
+      success = await updateHandler();
+    } else {
+      success = await createHandler();
+    }
+
+    if (success) {
+      toast({
+        description: 'Saved!',
+        status: 'success',
+        duration: 5000,
+      });
+      await getData();
+    }
+    setIsLoading(false);
+  };
 
   return (
-    <>
-      <Heading></Heading>
-    </>
+    <Container maxW="100%">
+      <Flex>
+        <Heading as="h3" mb={5}>
+          Leadership Team Manager
+        </Heading>
+
+        <Spacer />
+
+        <Button colorScheme="red" onClick={resetHandler} mr={2}>
+          RESET
+        </Button>
+        <Button
+          leftIcon={id !== '' ? <EditIcon /> : <AddIcon />}
+          colorScheme="teal"
+          onClick={onOpen}
+        >
+          {id !== '' ? 'EDIT' : 'ADD'}
```

Comment:
there's actually an annoying UX here, once you select any row on AgGrid, looks like i can't easily unselect it. Locking the button to forever be in edit state. I would need to refresh to be able to add a new entry if i wish to do so.

Reply:
<img width="220" alt="image" src="https://github.com/hmcc-global/hmcchk-web/assets/26876443/2de06b4e-f95a-4277-98aa-7e7f14a083e1">
oh yea you have to click the RESET button here to unselect it. i guess that's not very straightforward? or would a description be better or is there a better name for the button?

Reply:
note to self: 
- [ ] make the RESET button here more intuitive looking. 
- [ ] also make these buttons to be sticky so you can see it even when you scroll to the bottom of the table